### DISCUSSION TITLE
Connection details issues

### DISCUSSION THREAD

**Discussion by curcaandrei** [Erudika/scoold#446] [Q&A] [answered]
Hello,
I am trying to use both Para and Scoold locally and run them via Intellij.
I am running with this command so the configuration file is taken into account:
![image](https://github.com/Erudika/scoold/assets/42220021/e87c7230-7a1c-4ce9-88ed-4083b2895d4e)

I have this Para configuration:

```
para.app_name = "Para"
para.port = 8090
para.env = "production"
para.security.allow_unverified_emails = true
para.clients_can_access_root_app = true
para.cache_enabled = true
para.access_key = "app:scoold"
para.app_secret_key = "DjWNpEgJNeP4f6XALD3pRVB5k8KPkVUS"
para.cors_enabled = false
para.security.api_security = false
para.security.csrf_protection = false
para.cors_enabled = false
para.worker_id = 1
para.dao = "SqlDAO"
para.search = "LuceneSearch"

para.sql.driver = "oracle.jdbc.OracleDriver"
para.sql.url = "oracle:thin:...."
para.sql.user = "..."
para.sql.password = "...."
para.clients_can_access_root_app = true

para.security.ignored = ["/", "/images/**", "/styles/**", "/scripts/**", "/v1/_setup/**", "/v1/_health", "/v1/_config/options", "/v1/_me"]

para.root_access_key = "app:para"
para.root_secret_key = "iTzzqzzvcuDaQeFa72cUPzBFD3W2ZQC/GIt3E7N0yoUyZ3ejdzF9Zg=="
```

then in scoold i have the following:
```
scoold.env = "production"
scoold.app_name = "Scoold"
scoold.para_access_key = "app:scoold"
scoold.para_secret_key = "DjWNpEgJNeP4f6XALD3pRVB5k8KPkVUS"
scoold.para_endpoint = "http://localhost:8090"
scoold.admins = "mymail@gmail.com"

```

and i get:
`No connection to Para backend. Retrying connection in 10s (attempt 1 of 10)...`

if i use the root access and secret key, it works but i get: `403 - 403 - Can't authenticate user with app 'app:para' using provider 'password'. Reason: clients aren't allowed to access root app.`

After searching for a bit i've found that i need to create a child app ( i firstly though para.access_key would take care of it instead of para.root_access_key ). Then I've tried to use the cli i get:
![image](https://github.com/Erudika/scoold/assets/42220021/cd3e7d8f-7b7d-4d20-8491-45115dc3e672)
So then i've tried to use the endpoints:
![image](https://github.com/Erudika/scoold/assets/42220021/4cd524cb-d8a4-4f04-a3de-745b486c1eaa)
![image](https://github.com/Erudika/scoold/assets/42220021/14777181-46c2-478b-9ebf-cdeaa314aa5a)

What am I doing wrong?

**Comment by albogdano** [comment, marked as answer]
You might want to start over or at least change the secret keys for `app:para` and `app:scoold`.
Here's how the Para configuration should look like:
```
para.port = 8090
para.env = "production"
para.security.allow_unverified_emails = true
para.clients_can_access_root_app = true
para.cache_enabled = true
para.cors_enabled = false

# MAKE SURE THIS IS true IN PROD!
para.security.api_security = false

para.security.csrf_protection = false
para.cors_enabled = false
para.worker_id = 1
para.dao = "SqlDAO"
para.search = "LuceneSearch"

para.sql.driver = "oracle.jdbc.OracleDriver"
para.sql.url = "oracle:thin:...."
para.sql.user = "..."
para.sql.password = "...."
para.clients_can_access_root_app = true

para.root_access_key = "app:para"
para.root_secret_key = "iTzzqzzvcuDaQeFa72cUPzBFD3W2ZQC/GIt3E7N0yoUyZ3ejdzF9Zg=="
```

Run `para-cli setup` again and use the root secret key for `app:para` (from the file above), so enter:
Para Access Key: app:para
Para Secret Key: iTzzqzzvcuDaQeFa72cUPzBFD3W2ZQC/GIt3E7N0yoUyZ3ejdzF9Zg==
Endpoint: http://localhost:8090

Then create a new app called "Scoold", i.e. `app:scoold` from the Para CLI tool.
Then add the newly generated secret key to your Scoold configuration file as `para_secret_key`.
Restart both servers and you should be done.

If you see errors, delete the `./data` folder in the Para directory and reinitialize Para as described in the [Scoold README](https://github.com/Erudika/scoold?tab=readme-ov-file#quick-start-with-a-self-hosted-para-backend-harder)
